packages/media/src/webrtc/VegaRtcManager/utils.ts

```diff
@@ -0,0 +1,54 @@
+export function getLayers(
+    {
+        width,
+        height,
+    }: {
+        width: number;
+        height: number;
+    },
+    {
+        numberOfActiveVideos,
+        numberOfTemporalLayers,
+    }: {
+        numberOfActiveVideos: number;
+        numberOfTemporalLayers: number;
+    },
+) {
+    const maxSide = Math.max(width, height);
+    let spatialLayer = maxSide >= 480 ? (maxSide >= 960 ? 2 : 1) : 0;
+    let temporalLayer = numberOfTemporalLayers - 1; // default to full framerate
+
+    // if we are rendering tile-like sizes we reduce framerate by 50%
+    if (maxSide < 100) {
+        temporalLayer = Math.max(numberOfTemporalLayers - 2, 0);
+    }
+
+    // if we are rendering many videos, we reduce framerate by 50% for all videos with lowest spatial layer
+    if (numberOfActiveVideos > 8 && spatialLayer === 0) {
+        temporalLayer = Math.max(numberOfTemporalLayers - 2, 0);
+    }
+
+    // increase resolution if few videos (like 1:1 on phones)
+    // todo: consider spatialLayer 1 for maxSide > 200. 4 in a room is considerable higher quality on phones this way
+    // ... but might conflict with our intent of mobile-mode
+    if (numberOfActiveVideos < 4 && maxSide > 300 && spatialLayer === 0) {
+        spatialLayer = 1;
+    }
+
+    return { spatialLayer, temporalLayer };
+}
+
+export function getNumberOfActiveVideos(consumers: any) {
```

Comment:
What do you think about adding some types as we go?

these two functions are `Consumer[]` and `Consumer` imported from
`import { Consumer } from "mediasoup-client/lib/Consumer";`

Reply:
I can do that in a separate PR 👍

Reply:
I will do 👍

Reply:
Ok, Im not going to add this just now. Adding the types gave a few errors, as we are relying on private variables. I want us to set up an e2e test for stream resolution before I go ahead and change how we access some of this data.